### PR TITLE
5144 rename throttle component changed

### PR DESCRIPTION
**Pull request by diarmidmackenzie**
**Description:**

PR as per #5144

**Changes proposed:**
- name throttleComponentChanged -> throttleLeadingAndTrailing
- leave throttleComponentChanged utility function for 1.3.0 back-compatibility
- add documentation for throttleLeadingAndTrailing in utils API docs.

### CODE REVIEW

**Review by vincentfretin — COMMENTED** [review]
This looks good to me.

File: src/utils/index.js [hunk 1]
```diff
@@ -99,6 +106,12 @@ module.exports.throttleComponentChanged = function (functionToThrottle, minimumI
   };
 };
 
+/**
+ * Identical to throttleLeadingAndTrailing.
+ * Exists for back-compatibility with 1.3.0.
+ */
+module.exports.throttleComponentChanged = module.exports.throttleLeadingAndTrailing;
```
Comment (dmarcos) [inline src/utils/index.js:113]: I think we can just rename. Don't think many were relying on this. We can add a deprecation note in next release notes.

File: docs/core/utils.md [hunk 2]
```diff
@@ -212,6 +212,22 @@ AFRAME.registerComponent('foo', {
...
+
+Example use cases:
+
+ * synchronizing state based on the componentchanged event
+ * following a mouse pointer using the mousemove event
+ * integrating with [THREE.TransformControls](https://threejs.org/docs/#examples/en/controls/TransformControls), via the objectChange event.
+
+A context such as `this` can be provided to handle function binding for convenience. 
+
+The same as [lodash's`throttle`][lodash], with `leading` and `trailing` options both set to `true`
```
Comment (dmarcos) [inline docs/core/utils.md:229]: wonder if this reference is a liability long term in case lodash API changes.

Comment (diarmidmackenzie) [reply docs/core/utils.md:227]: Maybe, but the existing docs already reference loadash throttle (the same link)

See: https://aframe.io/docs/1.3.0/core/utils.html#aframe-utils-throttle-function-minimuminterval-optionalcontext

This doesn't make the situation any worse, I think.

**Comment by dmarcos** [comment]
Thank you!

**Pull request merged**